mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -1869,33 +1878,34 @@ impl ElectrumClient {
         rpc_func!(self, "blockchain.block.headers", start_height, count)
     }
 
-    pub fn retrieve_last_headers(
+    pub fn retrieve_headers(
         &self,
-        blocks_limit_to_check: NonZeroU64,
-        block_height: u64,
-    ) -> UtxoRpcFut<(HashMap<u64, BlockHeader>, Vec<BlockHeader>)> {
+        from: u64,
+        to: u64,
+    ) -> UtxoRpcFut<(HashMap<u64, BlockHeader>, Vec<BlockHeader>, u64)> {
         let coin_name = self.coin_ticker.clone();
-        let (from, count) = {
-            let from = if block_height < blocks_limit_to_check.get() {
-                0
-            } else {
-                block_height - blocks_limit_to_check.get()
-            };
-            (from, blocks_limit_to_check)
+        if from == 0 || to < from {
+            return Box::new(futures01::future::err(
+                UtxoRpcError::Internal("Invalid values for from/to parameters".to_string()).into(),
+            ));
+        }
+        let count: NonZeroU64 = match (to - from + 1).try_into() {
+            Ok(c) => c,
+            Err(e) => return Box::new(futures01::future::err(UtxoRpcError::Internal(e.to_string()).into())),
         };
         Box::new(
             self.blockchain_block_headers(from, count)
                 .map_to_mm_fut(UtxoRpcError::from)
                 .and_then(move |headers| {
-                    let (block_registry, block_headers) = {
+                    let (block_registry, block_headers, last_height) = {
                         if headers.count == 0 {
                             return MmError::err(UtxoRpcError::Internal("No headers available".to_string()));
                         }
                         let len = CompactInteger::from(headers.count);
                         let mut serialized = serialize(&len).take();
                         serialized.extend(headers.hex.0.into_iter());
-                        let coin_variant = coin_variant_by_ticker(&coin_name);
-                        let mut reader = Reader::new_with_coin_variant(serialized.as_slice(), coin_variant);
+                        let mut reader =
+                            Reader::new_with_coin_variant(serialized.as_slice(), coin_name.as_str().into());
                         let maybe_block_headers = reader.read_list::<BlockHeader>();
```

Comment:
A note, there is a macro called `drop_mutability` in `common` module. You can use it on such cases like this to prevent invalid data assignment to the mutable addresses.

For example:
```rs
let mut x = 10;
.
.
x = 1;
drop_mutability(x); // Since the variable no longer needs mutability
.
.
.
.
.
Ok(())
```

Reply:
Done